packages/realm-server/server.ts

```diff
@@ -665,6 +727,122 @@ export class RealmServer {
+        this.dbAdapter,
+        user.id,
+      );
+    }
+
+    let currentSubscriptionCycle: SubscriptionCycle | null | undefined =
+      undefined;
+    let plan: Plan | undefined = undefined;
+    if (mostRecentSubscription) {
+      [currentSubscriptionCycle, plan] = await Promise.all([
+        getMostRecentSubscriptionCycle(
+          this.dbAdapter,
+          mostRecentSubscription.id,
+        ),
+        getPlan(this.dbAdapter, mostRecentSubscription.planId),
+      ]);
+    }
+
+    let extraCreditsAvailableInBalance = await sumUpCreditsLedger(
+      this.dbAdapter,
+      {
+        creditType: ['extra_credit', 'extra_credit_used'],
+      },
+    );
+    let creditsAvailableInPlanAllowance: number | null = null;
+    if (currentSubscriptionCycle) {
+      creditsAvailableInPlanAllowance = await sumUpCreditsLedger(
+        this.dbAdapter,
+        {
+          creditType: ['plan_allowance', 'plan_allowance_used'],
+          subscriptionCycleId: currentSubscriptionCycle.id,
+        },
+      );
+    }
+
+    let responseBody = {
+      data: user
+        ? {
+            type: 'user',
+            id: user.id,
+            attributes: {
+              matrixUserId: user.matrixUserId,
+              stripeCustomerId: user.stripeCustomerId,
+              creditsAvailableInPlanAllowance: creditsAvailableInPlanAllowance,
+              extraCreditsAvailableInBalance: extraCreditsAvailableInBalance,
+            },
+            relationships: {
+              subscription: mostRecentSubscription
+                ? {
+                    data: {
+                      type: 'subscription',
+                      id: mostRecentSubscription.id,
+                    },
+                  }
+                : null,
+            },
+          }
+        : null,
+      included: mostRecentSubscription
+        ? [
+            {
+              type: 'subscription',
+              id: mostRecentSubscription.id,
+              attributes: {
+                startedAt: mostRecentSubscription.startedAt,
+                endedAt: mostRecentSubscription.endedAt ?? null,
+                status: mostRecentSubscription.status,
+              },
+              relationships: {
+                plan: {
+                  data: {
+                    type: 'plan',
+                    id: plan!.id,
+                  },
+                },
+              },
+            },
+            {
+              type: 'plan',
+              id: plan!.id,
+              attributes: {
+                name: plan!.name,
+                monthlyPrice: plan!.monthlyPrice,
+                creditsIncluded: plan!.creditsIncluded,
+              },
+            },
+          ]
+        : null,
+    } as FetchUserResponse;
+
+    return setContextResponse(
+      ctxt,
+      new Response(JSON.stringify(responseBody), {
+        headers: { 'content-type': SupportedMimeType.JSONAPI },
+      }),
+    );
+  };
+
```

Comment:
Maybe it’s time to break this file up somewhat, the collection of disparate route handlers invites separate files.

Reply:
Updated [7dea872](https://github.com/cardstack/boxel/pull/1756/commits/7dea872772f364d2806a1041aa635cd813896123).